File: python-client-sdk/src/orbapi/models/operations/get_coupons.py
from __future__ import annotations
import dataclasses
from ..shared import coupon as shared_coupon
from dataclasses_json import Undefined, dataclass_json
from orbapi import utils
from typing import Any, Optional


@dataclasses.dataclass
class GetCouponsQueryParams:
    redemption_code: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'redemption_code', 'style': 'form', 'explode': True }})
    show_archived: Optional[bool] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'show_archived', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetCouponsRequest:
    query_params: GetCouponsQueryParams = dataclasses.field()
    

@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class GetCoupons200ApplicationJSON:
    data: Optional[list[shared_coupon.Coupon]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data'), 'exclude': lambda f: f is None }})
    pagination_metadata: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pagination_metadata'), 'exclude': lambda f: f is None }})
    

@dataclasses.dataclass
class GetCouponsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    get_coupons_200_application_json_object: Optional[GetCoupons200ApplicationJSON] = dataclasses.field(default=None)
    

File: typescript-client-sdk/src/sdk/models/operations/getcoupons.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Expose, Type } from "class-transformer";


export class GetCouponsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=redemption_code" })
  redemptionCode?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=show_archived" })
  showArchived?: boolean;
}

export class GetCouponsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetCouponsQueryParams;
}

export class GetCoupons200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.Coupon })
  @Expose({ name: "data" })
  @Type(() => shared.Coupon)
  data?: shared.Coupon[];

  @SpeakeasyMetadata()
  @Expose({ name: "pagination_metadata" })
  paginationMetadata?: Record<string, any>;
}

export class GetCouponsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getCoupons200ApplicationJSONObject?: GetCoupons200ApplicationJSON;
}